swap_meet/vendor.py

```diff
@@ -1,2 +1,58 @@
+from operator import attrgetter
+
 class Vendor:
-    pass
+    def __init__(self, inventory=None):
+        self.inventory = inventory if inventory is not None else []
+    
+    def add(self, item_to_add):
+        self.inventory.append(item_to_add)
+        return item_to_add
+    
+    def remove(self, item_to_remove):
+        if item_to_remove in self.inventory:
+            self.inventory.remove(item_to_remove)
+            return item_to_remove
+        else:
+            return False 
+    
+    def get_by_category(self, category):
+        return [item for item in self.inventory if item.category == category]
+
+    def swap_items(self, other, my_item, their_item):
+        if my_item not in self.inventory or \
+            their_item not in other.inventory:
+            return False 
+        else: 
+            self.remove(my_item), other.add(my_item)
+            other.remove(their_item), self.add(their_item)
```

Comment:
we can take this a step further! `remove` returns the item, so let's pass it in to add!

```suggestion
            other.add(self.remove(my_item))
            self.add(other.remove(their_item))
```